source/server/server.cc

```diff
@@ -308,13 +307,12 @@ void InstanceImpl::initialize(Options& options,
 
   cluster_manager_factory_ = std::make_unique<Upstream::ProdClusterManagerFactory>(
       runtime(), stats(), threadLocal(), random(), dnsResolver(), sslContextManager(), dispatcher(),
-      localInfo(), secretManager(), api());
+      localInfo(), secretManager(), api(), http_context_);
 
   // Now the configuration gets parsed. The configuration may start setting thread local data
   // per above. See MainImpl::initialize() for why we do this pointer dance.
-  Configuration::MainImpl* main_config = new Configuration::MainImpl();
-  config_.reset(main_config);
-  main_config->initialize(bootstrap_, *this, *cluster_manager_factory_);
+  config_.initialize(bootstrap_, *this, *cluster_manager_factory_);
```

Comment:
I guess you decided to use the impl directly instead of the interface. Any particular reason for this? I guess it's an OK change. If we keep it can you change the comment up at line 313?

Reply:
Edited comment. The reason I changed it was it seemed simpler to directly contain the ConfigImpl -- the two-phase construct process happens either way but it didn't seem important when the first phase occurred.

Reply:
Let's discuss in the other PR.